lib/routes/routeMetadata.js

```diff
@@ -37,21 +37,35 @@ function routeMetadata(clientIP, request, response, log) {
     log.debug('routing request', { method: 'routeMetadata' });
     log.addDefaultFields({ clientIP, httpMethod: request.method });
     _normalizeMetadataRequest(request);
-    const requestContexts = prepareRequestContexts('objectReplicate', request);
+
+    // restrict access to only routes ending in bucket, log or id
+    const { resourceType, subResource } = request;
+    if (resourceType === 'admin'
+        && !['bucket', 'log', 'id'].includes(subResource)) {
+        return responseJSONBody(errors.NotImplemented, null, response, log);
+    }
+    const ip = request.headers['x-forwarded-for'] ||
+    request.socket.remoteAddress;
+    const requestContexts = [new RequestContext(request.headers, request.query,
+        request.generalResource, request.specificResource, ip,
+        request.connection.encrypted, request.resourceType, 'metadata')];
     return waterfall([
-        next => auth.server.doAuth(request, log, (err, userInfo) => {
+        next => auth.server.doAuth(request, log, (err, userInfo, authRes) => {
             if (err) {
                 log.debug('authentication error', {
                     error: err,
                     method: request.method,
                     bucketName: request.bucketName,
                     objectKey: request.objectKey,
                 });
+                return next(err);
+            }
+            if (authRes && !authRes.isAllowed) {
```

Comment:
Looking at cloudserver lib/api/api.js, the returned `authRes` there is an array, I'm assuming it's because there is a 1:1 mapping between input requestContexts and output authorization results, so it seems the proper check would be `authRes && authRes[0].isAllowed`

Reply:
Ah, you are right. I overlooked the response.